src/util/signature.cr

```diff
@@ -48,4 +48,30 @@ class Dir
     end
     Digest::CRC32.checksum(signatures.sort.join).to_u64
   end
+
+  # Returns the contents signature of the directory at dirname for checking
+  #   to rescan.
+  # Rescan conditions:
+  #   - When a file added, moved, removed, renamed (including which in nested
+  #       directories)
+  def self.contents_signature(dirname, cache = {} of String => String) : String
+    return cache[dirname] if cache[dirname]?
+    signatures = [] of String
+    self.open dirname do |dir|
+      dir.entries.sort.each do |fn|
+        next if fn.starts_with? "."
+        path = File.join dirname, fn
+        if File.directory? path
+          signatures << Dir.contents_signature path
```

Comment:
Why aren't we passing `cache` to `contents_signature` here?

Reply:
Oh, right ;)

Reply:
Thanks!